### PR TITLE
Hooks refactor

### LINKED ISSUE #1542: Replace all model/repo code with hooks
## Description
We've begun moving away from the class based model/repo classes, and all new features are written using hooks instead, but old features are still using models.

## Relevant Job Stories
None

## Estimated size (S, M, L)
I think this issue is large because it requires significant refactoring and being somewhat comfortable with writing hooks, good patterns etc.

## Prerequisites
None

## Requirements
- [ ] Replace all model/repo code with hooks
- [ ] Find ways of reusing logic

DONE
calendar        
files               
smartSearch 
user               

IN REVIEW
organizations
tags
tasks
callAssignments

IN PROGRESS
views           (Haeju)
campaigns  (Ziggi) (models, react-query)

TODO
events          in progress (Rebeca)
journeys       not started (react-query)
profile          not started (react-query)
search          not started (react-query)
surveys         not started (models)

## Possible implementations
Unknown

## Design specifications
None

## Open questions
None

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR begins the work of unifying how components get data, from the Model/Repo approach we have been testing, to an approach using only hooks. 

## Changes
* Changes CallAssignments, Tags, Tasks and Organizations 

## Notes to reviewer
Provoke for bugs where you can 

## Related issues
Partly resolves #1542

### CODE REVIEW

**Comment by richardolsson** [comment]
Here are my notes from our call the other day. Dumping them here so that I can get rid of them locally. 😊 

```
calendar        done
files           done
smartSearch     done
user            done

views           in progress (Haeju)
tasks           in review (Haeju) 
callAssignments in review (Ziggi)
tags            changes required (Haeju)
organizations   changes required (Rebeca)

events          in progress (Rebeca)
campaigns       not started (models, react-query)
journeys        not started (react-query)
profile         not started (react-query)
search          not started (react-query)
surveys         not started (models)
```

File: src/features/callAssignments/hooks/useCallAssignmentStats.ts [hunk 1]
```diff
@@ -126,8 +126,10 @@ export default function useCallAssignmentStats(
   };
 
   return {
-    ...getStats(),
+    data: getStats().data,
+    error: getStats().error,
```
Comment (rebecarubio) [inline src/features/callAssignments/hooks/useCallAssignmentStats.ts:130]: For me all this calls to `getStats()` look unnecessary. Is it possible to save the return of the function in a variable and work with that in the return? Like stats.data, stats.error or event just return the future object?

File: src/features/callAssignments/hooks/useCallers.ts [hunk 2]
```diff
@@ -138,7 +138,9 @@ export default function useCallers(
 
   return {
     addCaller,
-    ...getFilteredCallers(),
+    data: getFilteredCallers().data,
+    error: getFilteredCallers().error,
+    isLoading: getFilteredCallers().isLoading,
```
Comment (rebecarubio) [inline src/features/callAssignments/hooks/useCallers.ts:143]: Same here from my comment above

**Review by rebecarubio — CHANGES_REQUESTED** [review]
Good job refactoring this! My only concern as far as I can see it's the return of some functions, as I commented. I think that would fix some of the issues that @kaulfield23 has detected, like the disabled save in call instructions.

**Comment by kaulfield23** [comment]
Nice 👍 ! I've tested every hooks and its return values on web to see if everything works well. I have found a problem with call instructions that can't be updated. About codes, I also agree on Rebeca's comments! We assigned future values to a variable and returned that variable from hook. You can find those patterns in our codes!

**Comment by richardolsson** [comment]
Once everything else has been merged into this, and before this is ready to merge with `main`, I think there are some additional cleaning tasks that could take place. I'm adding them here to remember, feel free to add to the list (or postpone them for later if you prefer):

- [ ] Get rid of `useModel()`, `ModelBase` and similar code
- [ ] Remove the `react-query` dependency
- [ ] Get rid of `ZUIQuery`

**Comment by richardolsson** [comment]
Here's my analysis of the current situation.

Places that still rely on `react-query`:

- [ ] Breadcrumbs (I will fix)
- [ ] `ZUIPersonSelect` (I will fix)
- [ ] Search (I will fix)
- [ ] Profile page (I will fix)
- [ ] `EditCell` in `SurveySubmissionsList` (I will fix)
- [ ] `CampaignDetailsForm` (maybe you can investigate and fix, @ziggabyte?)
- [x] Journeys (being worked on by @ziggabyte)
- [x] Views (being worked on by @kaulfield23)
- [x] Smart Search (being worked on by @rebecarubio)


Places that still rely on models:
- [ ] `CallAssignmentStatusCard` (will you investigate, @ziggabyte?)
- [x] Views (being worked on by @kaulfield23)

**Comment by kaulfield23** [comment]
These things are need to be fixed
- A bug where a task can't be published when it has assigned targets by using smart search.
: line 96 to 101 in `src\pages\organize\[orgId]\projects\[campId]\calendar\tasks\[taskId]\assignees.tsx` needs to be fixed

- `UpdateColumnOrder` in `main` branch
: `UpdateColumnOrder` in `src\features\views\models\ViewDataModel.ts` and its relevant codes does not exist in this branch.

**Comment by richardolsson** [comment]
Once #1603 is merged, I believe everything has been solved except this:

>  Address the situation in FetchAPIClient, line 94

This is not really a problem related to hooks/models/etc, it works ok without changes, and can be worked on at a later time. I can't really think of a great solution right now, so I'd like to postpone it and just leave the `TODO` comment that is already there.

**Comment by richardolsson** [comment]
I've resolved all merge conflicts now. In the process I found a bug that I reported as #1604 and another that I fixed in d36b2ad.

**Review by richardolsson — APPROVED** [review]
I don't see how reviewing this huge PR is ever going to be feasible, so I'm just going to approve it based on the fact that all the individual sub-PRs that have lead up to it have been approved.

Let's merge!

**Pull request merged**